src/filesystem/sharing/shared_file.rs

```diff
@@ -0,0 +1,66 @@
+use base64::{engine::general_purpose::URL_SAFE, Engine as _};
+use serde::{Deserialize, Serialize};
+use wnfs::{common::dagcbor, libipld::Cid, private::share::SharePayload};
+
+use super::SharingError;
+
+#[derive(Debug, Clone)]
+pub struct SharedFile {
```

Comment:
This includes the content_cid, snapshot_key, and label HashOutput internal to the SharePayload. 
The forest_cid references the serialized IPLD version of the forest. If we can't grab that Block from somewhere, then we presumably cant grab any of the blocks it references either. so this is my idea of a more space-efficient export than just actually exporting the forest with it. 
that said, there are no block locations in this struct. should there be?

Reply:
It's kind of a trade off whether blocks should be included... Knowing them ahead of time means we can proactively pull them all at once in parallel, this is a slower way as we have to retrieve the blocks associated with the forest, decrypt it then retrieve the blocks one at a time (there is some possibility for parallelism here once the forest block is decoded).

That being said files can have A LOT of blocks and I don't think we could fit all of those CIDs into a URL... So yeah I think we'll have to stick with the slow path for now (no changes needed).